tasks/homework/homework.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"fmt"
+)
+
+type Animal struct {
+	ID   int
+	Name string
+}
+
+type Areas map[string]Area
+
+type Area struct {
+	Name    string
+	Type    string
+	Sectors map[string]Sector
+}
+
+const (
+	SectorTypeTechnical = "technical"
+	SectorTypeAnimals   = "animals"
+)
+
+type Sector struct {
+	Subtype string
+	Animals []Animal
+}
+
+func (s *Sector) AddAnimal(animal Animal) {
+	s.Animals = append(s.Animals, animal)
+}
+
+func (s *Sector) MoveAnimal(animalID int, distSector *Sector) {
+	for i, animal := range s.Animals {
+		if animal.ID == animalID {
+			distSector.AddAnimal(animal)
+			s.Animals = append(s.Animals[:i], s.Animals[i+1:]...)
+			return
+		}
+	}
+}
+
+func (s *Sector) CleanUp() {
+	if s.Subtype == SectorTypeTechnical {
+		fmt.Println("technical sector cleaned up")
+	}
+}
+
+func (s *Sector) Feed(animal *Animal) {
+	if s.Subtype == SectorTypeTechnical {
+		fmt.Printf("feeding the %s\n", animal.Name)
+	}
+}
+
+type Zoo struct {
+	Areas Areas
+}
+
+func main() {
+	z := Zoo{
+		Areas: buildAreas(),
+	}
+
+	fmt.Println("Try to find by name Eagle")
+	eagle, err := z.FindAnimalByName("Eagle")
+	if nil == err {
```

Comment:
Ось така конструкція завжди в Go коді має бути:
```suggestion
	if err == nil {
```

Навряд чи можна зустріти `nil == err` в проектах з хорошим кодом.